pages/stores/index.js

```diff
@@ -3,26 +3,55 @@ import Layout from '../../components/layout'
 import Navbar from '../../components/navbar'
 import { StoreCard } from '../../components/store/card'
 import { getStores } from '../../data/stores'
+import { getProductByCustomerId, getProducts } from '../../data/products'
+import { ProductCard } from '../../components/product/card'
+import { IndexStoreCard } from '../../components/store/index-card'
 
 
 export default function Stores() {
   const [stores, setStores] = useState([])
+  const [products, setProducts] = useState([])
 
   useEffect(() => {
-    getStores().then(data => {
+    let activeStores = []
+    getStores().then(stores => {
+      if (stores) {
+        for (let store of stores) {
+          getProductByCustomerId(store.owner.id).then(data => {
+            if (data.length != 0)
+            store.productcount = data.length
+            activeStores.push(store)
+          })
+        }
+        setStores(activeStores)
+      }
+    })
+  }, [])
+
+  useEffect(() => {
+    getProducts().then(data => {
       if (data) {
-        setStores(data)
+        setProducts(data)
       }
     })
   }, [])
 
   return (
     <>
       <h1 className="title">Stores</h1>
-      <div className="columns is-multiline">
+      <div>
       {
         stores.map(store => (
-          <StoreCard store={store} key={store.id} />
+          <>
+          <div className="columns is multi-line">
+            <IndexStoreCard store={store} key={store.id}/>
+          </div>
+          <div className="columns is multi-line">
+          {products.filter((product) => (product.customer_id==store.owner.id)).slice(0,4).map((product) => (
```

Comment:
This works nicely! 

Food for thought - there is a way to update the StoreSerializer used in the GET /stores method that would put a list of each store's products on the store object that is returned by getStores(), as well as the already-calculated product_count, similar to what happens with GET /stores/1. Let me know if you are curious to see and discuss that alternate method. Otherwise, your method works great! Just mentioning in case you want to see a more Django-driven filtering approach.